rotkehlchen/db/evmtx.py

```diff
@@ -271,14 +272,18 @@ def get_transaction_hashes_not_decoded(
             self,
             chain_id: Optional[ChainID],
             limit: Optional[int],
+            addresses: Optional[List[ChecksumEvmAddress]],
     ) -> List[EVMTxHash]:
         """Get transaction hashes for the transactions that have not been decoded.
         Optionally by chain id.
         If the limit argument is provided then it is used in the SQL query with
         the default order.
         """
         querystr = 'SELECT A.tx_hash from ' + TRANSACTIONS_MISSING_DECODING_QUERY
-        bindings = []
+        bindings: List[Union[ChecksumEvmAddress, int]] = []
+        if addresses is not None:
+            bindings += addresses + addresses
+            querystr += f' AND (C.from_address IN ({",".join("?"*len(addresses))}) OR C.to_address IN ({",".join("?"*len(addresses))}))'  # noqa: E501
         if chain_id is not None:
```

Comment:
Since here filtering happens by many parameters (not just `addresses`) then maybe will make sense to use a proper filter class here?